source/slang/slang-state-serialize.cpp

```diff
@@ -1072,10 +1074,234 @@ struct LoadContext
+
+        switch (SlangLineDirectiveMode(requestState->lineDirectiveMode))
+        {
+            case SLANG_LINE_DIRECTIVE_MODE_DEFAULT: break;
+            case SLANG_LINE_DIRECTIVE_MODE_NONE:
+            {
+                cmd.addArg("-line-directive-mode none"); break;
+            }
+            default: break;
+        }
+
+        switch (SlangDebugInfoLevel(requestState->debugInfoLevel))
+        {
+            case SLANG_DEBUG_INFO_LEVEL_STANDARD:       cmd.addArg("-g"); break;
+            case SLANG_DEBUG_INFO_LEVEL_NONE:           cmd.addArg("-g0"); break;
+            case SLANG_DEBUG_INFO_LEVEL_MINIMAL:        cmd.addArg("-g1"); break;
+            case SLANG_DEBUG_INFO_LEVEL_MAXIMAL:        cmd.addArg("-g3"); break;
+            default: break;
+        }
+
+        switch (SlangOptimizationLevel(requestState->optimizationLevel))
+        {
+            case SLANG_OPTIMIZATION_LEVEL_NONE:         cmd.addArg("-O0"); break;
+            case SLANG_OPTIMIZATION_LEVEL_DEFAULT:      cmd.addArg("-O");  break;
+            case SLANG_OPTIMIZATION_LEVEL_HIGH:         cmd.addArg("-O2"); break;
+            case SLANG_OPTIMIZATION_LEVEL_MAXIMAL:      cmd.addArg("-O3"); break;
+            default: break;
+        }
+
+        //spSetOutputContainerFormat(externalRequest, SlangContainerFormat(requestState->containerFormat));
+
+        switch (SlangPassThrough(requestState->passThroughMode))
+        {
+            case SLANG_PASS_THROUGH_NONE: break;
+            default:
+            {
+                cmd.addArg("-pass-through");
+                cmd.addArg(getPassThroughName(SlangPassThrough(requestState->passThroughMode)));
+                break;
+            }
+        }
+
+        //request->getBackEndReq()->useUnknownImageFormatAsDefault = requestState->useUnknownImageFormatAsDefault;
+        //request->getBackEndReq()->obfuscateCode = requestState->obfuscateCode;
+        //request->getFrontEndReq()->obfuscateCode = requestState->obfuscateCode;
+
+        switch (requestState->defaultMatrixLayoutMode)
+        {
+            case SLANG_MATRIX_LAYOUT_ROW_MAJOR:     cmd.addArg("-matrix-layout-row-major"); break;
+            case SLANG_MATRIX_LAYOUT_COLUMN_MAJOR:  cmd.addArg("-matrix-layout-column-major"); break;
+            default: break;
+        }
+    }
+
+    // Add the target requests
+    {
+        for (Index i = 0; i < requestState->targetRequests.getCount(); ++i)
+        {
+            TargetRequestState& src = base.asRaw(requestState->targetRequests[i]);
+
+            cmd.addArg("-target");
+            cmd.addArg(getCodeGenTargetName(CodeGenTarget(src.target)));
+
+            cmd.addArg("-profile");
+            cmd.addArg(Profile(src.profile).getName());
+
+            if (src.targetFlags & SLANG_TARGET_FLAG_PARAMETER_BLOCKS_USE_REGISTER_SPACES)
+            {
+                cmd.addArg("-parameter-blocks-use-register-spaces");
+            }
```

Comment:
FWIW I believe that flag is effectively deprecated. Whether or not parameter blocks use registers spaces is now determined by whether the target supports register spaces.

I'd mistakenly thought that SM 5.1 (which introduced spaces) was usable across both D3D11 and D3D12, so I added this option as a way to control whether or not register spaces would be used for SM 5.1. It turns out that SM 5.0 and below don't support spaces at all (so the flag isn't needed), while SM 5.1 and up only work with D3D12, so that using spaces freely isn't actually an issue.